### PR TITLE
Boss capture point and precache restructure

### PR DESCRIPTION
**Pull request by Trildar**
Implementing #1521. I'm currently assuming the capture pauses when it's contested rather than resetting. Is that correct? Also, how big should the capture point be?

Current state:
Spawns on death of boss. Seems to correctly animate to indicate capturing.
Does not yet actually do anything. Probably want to change how and when the colour changes to better indicate capturing team.

Also, the main ring particle (the only one that shows when the CP is empty) seems quite flaky. It flickers a bunch for some reason. No idea if it's just my computer. Would be great if someone could check that for me.

Credits to @carlosrpg for hacking together the initial particle.

### CODE REVIEW

**Comment by Trildar** [comment]
Also made separate precache list file since the list of things we're precaching is getting pretty long.

File: game/scripts/vscripts/modifiers/modifier_boss_capture_point.lua [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+    -- Controls how much of the dial to spawn. 1 is the full circle
+    ParticleManager:SetParticleControl(self.captureClockEffect, 11, Vector(0, 0, 1))
+  end
+  ParticleManager:SetParticleControl(self.captureClockEffect, 3, self:GetColor())
+end
+
+function modifier_boss_capture_point:GetColor()
+  local neutralColor = Vector(255, 150, 0)
+  local radiantColor = Vector(0, 162, 255)
+  local direColor = Vector(5, 155, 255)
```
Comment (carlosrpg) [inline game/scripts/vscripts/modifiers/modifier_boss_capture_point.lua]: The dire color is wrong in think. This is should be (241, 5, 5 ) or some sort of red

**Comment by warpdragon** [comment]
image
soemthing like this

**Comment by carlosrpg** [comment]
I figure out how to fix the particle ring problem, and how to attach the control point.
Still I could not figure out how to send a callback to the modifier.

**Comment by Trildar** [comment]
Ready for merge. @l34Um1 Current size is 300 units radius and the capture pauses without resetting when contested. It also does not care about units taking damage. Is that all as intended?

**Comment by l34Um1** [comment]
It should reset. Ideally it goes down at half the speed it went up when nobody is capturing. Size is cool, saw the video, looks great

**Comment by Trildar** [comment]
So it does a slow reset both when contested and when there's nobody in the capture point?

**Comment by l34Um1** [comment]
No when it is contested, meaning both teams have players in the circle it stays at the same it was before. If the circle is empty it slowly goes down

**Comment by Trildar** [comment]
And when it's being captured by the opposite team from previous? e.g. Radiant was inside for 5 seconds, then left, then Dire stands in it.

**Comment by l34Um1** [comment]
In that case it should go down faster. Maybe 1.5 the regular speed. Once the team starts capturing it for themselves it goes back to normal speed though

**Comment by Chronophylos** [comment]
I'm having problems displaying the Particle in the Asset Browser

**Comment by chrisinajar** [comment]
`    game/scripts/vscripts/components/boss/ai.lua:82:10: (W113) accessing undefined variable IsPlayerTeam`

**Pull request merged**